scalafix-core/src/main/scala/scalafix/rewrite/NoExtendsApp.scala

```diff
@@ -14,18 +14,24 @@ case class NoExtendsApp(mirror: Mirror) extends SemanticRewrite(mirror) {
   override def rewrite(ctx: RewriteCtx): Patch = {
     def wrapBodyInMain(template: Template) =
       ctx
-        .templateBodyTokens(template)
-        .map { body =>
+        .templateBodyTokens(template) match {
+        case None =>
+          ctx.reporter.warn(
+            "Missing or empty body for object that extends scala.App. See http://dotty.epfl.ch/docs/reference/dropped/delayed-init.html for possible workarounds.",
+            template.pos
+          )
+          Patch.empty
```

Comment:
I've considered adding a `Message(Patch, msg: String, Severity, Position)` patch that can be used to attach a message to a patch. We could then expose an api like `ctx.warn(pos, msg) = Message(Patch.empty, msg, warn, pos)`. We would then make `ctx.reporter` private and issue the warnings at the end of the world, sorted by position location.

I think that would make more sense than the current `warn(msg): Unit` signature. What do you think?